src/js/player.js

```diff
@@ -1903,6 +1903,13 @@ class Player extends Component {
         // the tech loop to check for a compatible technology
         this.sourceList_([source]);
       } else {
+        let previous = this.currentSrc();
+
+        this.trigger('beforesourcechange', {
```

Comment:
I was going to ask if this should be triggered on line 1888 (and I do think that it probably should) but given how recursive `src` is, we probably don't want to move this up.